banzai_floyds/tests/test_e2e.py

```diff
@@ -133,6 +135,61 @@ def test_if_arc_frames_were_created(self):
                 assert os.path.exists(expected_file)
 
 
+@pytest.mark.e2e
+@pytest.mark.fringe
+class TestFringeCreation:
+    @pytest.fixture(autouse=True)
+    def process_lampflat_frames(self):
+        logger.info('Reducing individual frames')
+
+        exchange = Exchange(os.getenv('FITS_EXCHANGE', 'fits_files'),
+                            type='fanout')
+        test_data = ascii.read(DATA_FILELIST)
+        with Connection(os.getenv('FITS_BROKER')) as conn:
+            producer = conn.Producer(exchange=exchange)
+            for row in test_data:
+                if 'w00.fits' in row['filename']:
+                    archive_record = requests.get(f'{os.getenv("API_ROOT")}frames/{row["frameid"]}').json()
+                    archive_record['frameid'] = archive_record['id']
+                    producer.publish(archive_record)
+            producer.release()
+
+        celery_join()
+        logger.info('Finished reducing individual frames')
+
+    @pytest.fixture(autouse=True)
+    def stack_flat_frames(self):
+        logger.info('Stacking Lamp Flats')
+        for site in ['ogg', 'cpt']:
+            runtime_context = dict(processed_path='/archive/engineering', log_level='debug', post_to_archive=False,
+                                   post_to_opensearch=False, fpack=True, reduction_level=91,
+                                   db_address=os.environ['DB_ADDRESS'], opensearch_qc_index='banzai_qc',
+                                   opensearch_url='https://opensearch.lco.global',
+                                   no_bpm=False, ignore_schedulability=True, use_only_older_calibrations=False,
+                                   preview_mode=False, max_tries=5, broker_url=os.getenv('FITS_BROKER'),
+                                   no_file_cache=False)
+            for setting in dir(settings):
+                if '__' != setting[:2] and not isinstance(getattr(settings, setting), ModuleType):
+                    runtime_context[setting] = getattr(settings, setting)
+
+            observations = {'request': {'configuration': {'LAMPFLAT': {'instrument_configs': {'exposure_count': 1}}}}}
+            instruments = dbs.get_instruments_at_site(site, runtime_context['db_address'])
+            for instrument in instruments:
+                if 'FLOYDS' in instrument.type:
+                    instrument_id = instrument.id
+                    break
+            stack_calibrations('2000-01-01', '2100-01-01', instrument_id, 'LAMPFLAT',
```

Comment:
Not sure if this matters here, but you might want to catch the exception in the case that FLOYDS is not in the Instrument type and therefore no `instrument_id` is found.

Reply:
I think since this is not really something we are exposing as a library call this is fine. If the db is not setup correctly we kind of want it to crash hard.